### PR TITLE
chore: create initial github pages site

### PR DESCRIPTION
**Pull request by mrderyk**
## CONTEXT

The repository can benefit from a page to easily demonstrate the abilities of the ReactSearch component. This page should pull in the latest version of component, and because of this can double as the dev environment.

## CHANGES
- Added more content around dev env page so it can be used as a github pages demo
- Add an action to deploy an updated github page on merging to main.
- Refactored dev build script and configs to allow the GH action to leverage these during GH pages deployment.
- Add a prettierrc file (needed this to constrain the string lengths on the page's `index.tsx` file)

## SCREENSHOTS
Please see the page [here](https://vectara.github.io/react-search/)
Note that prior to merging, I'll change the target branch to main. For the purpose of reviewing this PR, I'm having the action deploy a page on updates to this PR branch.

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -7,17 +7,11 @@ import {
   useRef,
   useEffect,
   useMemo,
-  ReactNode,
+  ReactNode
```
Comment (mrderyk) [inline src/index.tsx:10]: Changes here are related to introducing the prettier config file

File: dev/index.tsx [hunk 2]
```diff
@@ -1,21 +1,120 @@
...
+              Try it out below! You can also make changes to the options in the Configuration section to modify the
+              component's behavior/appearance.
+            </p>
+          </div>
+          <div className="searchOuterWrapper">
+            <div className="searchInnerWrapper">
+              <ReactSearch
+                corpusId={corpusId === "" ? DEFAULT_CORPUS_ID : corpusId}
+                customerId={customerId === "" ? DEFAULT_CUSTOMER_ID : customerId}
+                apiKey={apiKey === "" ? DEFAULT_API_KEY : apiKey}
```
Comment (mrderyk) [inline dev/index.tsx:53]: Here we ensure that if the field is blank, we use the default props that point to the docs page. This ensures that we don't voluntarily display the docs corpus details in the text fields.

Comment (cjcenizal) [reply dev/index.tsx:54]: Might be worth leaving this as a code comment.

**Review by cjcenizal — APPROVED** [review]
LGTM!

**Pull request merged**